### PR TITLE
build: Create r2 artefact instead of Docker

### PR DESCRIPTION
**Pull request by istarkov**
### Time of building the same project 
new:
image

was:
image

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: .github/workflows/cli-r2.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      - name: pnpm deploy
+        run: pnpm --filter 'webstudio-cloudlfare-template' deploy "${{ github.workspace }}/../cloudlfare-template"
+
+      - name: cleanup cloudflare template
+        run: |
+          find . -name 'typescript@*' -type d -prune -exec rm -rf '{}' + && \
+          find . -name '@storybook*' -type d -prune -exec rm -rf '{}' + && \
+          find . -name 'eslint@*' -type d -prune -exec rm -rf '{}' + && \
+          find . -name '@types+node@*' -type d -prune -exec rm -rf '{}' + && \
+          find . -name '@cloudflare+workers-types@*' -type d -prune -exec rm -rf '{}' +
```
Comment (kof) [inline .github/workflows/cli-r2.yaml:55]: this looks really nasty,  will potentially get out of sync as we move forward

File: .github/workflows/cli-r2.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          ref: ${{ github.sha }} # HEAD commit instead of merge commit
+
+      - uses: pnpm/action-setup@v2.2.4
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18.x"
+          cache: pnpm
+
+      - name: pnpm instal
```
Comment (kof) [inline .github/workflows/cli-r2.yaml:40]: use --prod flag to avoid dev dependenices

File: .github/workflows/cli-r2.yaml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  statuses: write # This is required for the GitHub Script createCommitStatus to work
+  packages: write
+
+jobs:
+  build:
+    env:
+      COMPATIBILITY_DATE: 2024-04-10
+
+    environment:
+      name: "staging"
```
Comment (TrySound) [inline .github/workflows/cli-r2.yaml:24]: 🤦

File: .github/workflows/cli-r2.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+          cache: pnpm
+
+      - name: pnpm instal
+        run: pnpm install --ignore-scripts
+
+      - name: pnpm build
+        run: pnpm --filter 'webstudio-cloudlfare-template^...' run build
+
+      - name: pnpm deploy
+        run: pnpm --filter 'webstudio-cloudlfare-template' deploy "${{ github.workspace }}/../cloudlfare-template"
```
Comment (TrySound) [inline .github/workflows/cli-r2.yaml:47]: is it homedir thingy?

**Review by TrySound — APPROVED** [review]

**Pull request merged**